lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -710,18 +994,97 @@ bool BootloaderCom::compareExpectedAndReceivedResponse(const uint8_t command[],
             return false;
         }
     }
-    else if(command == Zumo32U4Specification:: WRITE_MEMORY_PAGE)
+
+    else if(m_cmdProvider->iswriting == true)
+    {
+        uint8_t idx= temprovider->getCount();
+        LOG_INFO("Writing Data  %d to Zumo was successfull", idx);
+        return true;
+    }
+    else if(command == Zumo32U4Specification:: READ_MEMORY_PAGE)
     {
-        if (0 == memcmp(receivedResponse,Zumo32U4Specification::SET_MEMORY_ADDR,expectedSize))
+        if (m_firmwareFile)
+        {
+            /*File is already opened.*/
+        }
+        else
+        {
+            m_firmwareFile = LittleFS.open(m_fileName, "r");
+        }
+
+        /*Obtain a ByteCheckCmdProvider pointer from the current command provider.*/
+        ByteCheckCmdProvider* temprovider = (ByteCheckCmdProvider*)m_cmdProvider;
+        /*Get the count from the ByteCheckCmdProvider instance.*/
+        uint8_t idx = temprovider->getCount();
+ 
+        /*Fill the buffer with Fillvalues 0xFF.*/
+        if( nullptr != mybuffer)

```

Comment:
how can mybuffer (a global) ever be null ?

Reply:
It will be adjusted next time.